src/index.ts

```diff
@@ -30,11 +30,15 @@ export interface Options {
 const customInject = /([ \t]*)<!--__vite-plugin-inject-preload__-->/i
 
 export default function VitePluginInjectPreload(options: Options): Plugin {
+  let basePath = "/";
   return {
     name: 'vite-plugin-inject-preload',
     apply: 'build',
     transformIndexHtml: {
       enforce: 'post',
+      configResolved(resolvedConfig) {
+        if(resolvedConfig.base) basePath = resolvedConfig.base
```

Comment:
Not a big fan to load like that the resolved config because of the different type of string said in the doc https://vitejs.dev/config/shared-options.html#base

Reply:
ViteJS is so easy to use : the base path is already handle by the bundler so my comment above is not valid.